### ISSUE TITLE
Need Windows DNS configuration tests

### ISSUE THREAD

**Issue by PatrickLang** [kubernetes/kubernetes#73414]
We need to add Windows-specific DNS tests to cover the following fields:

  - Test cases needed for `dnsPolicy`: Default, ClusterFirst, None
  - Test cases needed for `dnsConfig`
  - Test cases needed for `hostname`
  - Test cases needed for `subdomain`

These need to run on Windows containers because the domain resolution policy is different, so different results need to be checked. Windows assumes that any name with a `.` in it is a FQDN, not a PQDN. 

There is an [open PR](https://github.com/kubernetes/kubernetes/pull/72729) to cover simple PQDN resolution, but they don't cover the Pod DNS configurations above. 

/sig windows
/milestone v1.14

**Comment by michellemcone** [comment]
FYI I'm working on this. I plan to start out by porting over [this test](https://github.com/kubernetes/kubernetes/blob/bb9a12d6f8875bed8e28d10a4351c96ab18f8247/test/e2e/network/dns.go#L254) to Windows.

**Comment by PatrickLang** [comment]
I just moved it into the in progress column. THanks!

**Comment by michellemcone** [comment]
@PatrickLang 

What test cases are needed for `dnsConfig`, `hostname`, and `subdomain`? My thoughts were that [this test](https://github.com/kubernetes/kubernetes/blob/bb9a12d6f8875bed8e28d10a4351c96ab18f8247/test/e2e/network/dns.go#L254) covers `dnsPolicy=None` and `dnsConfig`. I'm adding test cases for Default and ClusterFirst. 

Does [this test](https://github.com/kubernetes/kubernetes/blob/bb9a12d6f8875bed8e28d10a4351c96ab18f8247/test/e2e/network/dns.go#L125) cover the test cases needed for `hostname` and `subdomain` or do I need more? Or should I just be porting over the entire dns.go file?

**Comment by PatrickLang** [comment]
For the first one [should support configurable pod resolv.conf](https://github.com/kubernetes/kubernetes/blob/bb9a12d6f8875bed8e28d10a4351c96ab18f8247/test/e2e/network/dns.go#L254) - yes. The main differences will be checking the output of `ipconfig` instead of `cat /etc/resolv.conf`, and maybe using `nslookup` instead of `dig` but the flow is the same

Checking second one now

**Comment by PatrickLang** [comment]
Agree on second one. I think that will cover it. It may make sense to simplify it a bit because the common methods used there are all based on running bash + dig. If it works with busybox, that may be feasible.

The description at https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields matches those test cases

**Comment by daschott** [comment]
If at all possible, could we use native Windows cmdlets such as `resolve-dnsname` instead of relying on Linux cmdlets that have been ported to Windows like `dig`? Am not 100% confident if they are going to behave consistently.

**Comment by PatrickLang** [comment]
yeah, I think what David's suggesting makes sense if the test cases are forked for Windows.

**Comment by michellemcone** [comment]
Opened PR for enabling custom DNS on Azure CNI: https://github.com/Azure/azure-container-networking/pull/305

**Comment by PatrickLang** [comment]
/milestone clear
This will need to merge after 1.14 as we're waiting on CNI updates for these tests to pass. I added this status note to the Windows kep
